testsuite/config_reader.py

```diff
@@ -27,6 +27,19 @@ def get_vendor_config(vendor, override=''):
     if override:
         config = deep_merge(config, yaml.safe_load(override))
 
+    # We need to figure out which version the security calls needs to use.
+    # Interrogate configuration here.
+    try:
+        config['security_version'] = config['use_cases']['security']
+    except KeyError as error:
+        raise Exception("Bad Configuration, could not find version"
+                        "for security under 'use_cases'. (%s)" % error)
+
+    # The config file has the auth and api hierarchies under the version key.
+    # For ease of use, repoint the top level auth and api to be the relevant version.
+    config["auth"] = config["auth"]["versions"][config['security_version']]
```

Comment:
This kind of rewriting is a little bit magical -- it breaks the correspondance between what you see/type, and what's visible to the python code. It might be better to just write accessor functions to handle this, or to populate a new property...

Reply:
Made this explicit in a new property to refer to a "versioned auth" and "versioned api".